### PR TITLE
[3.11] gh-115198: Fix test_check_metadata_deprecation in test_distutils

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-115198

### CODE REVIEW

File: Lib/distutils/tests/test_register.py [hunk 1]
```diff
@@ -291,6 +291,7 @@ def test_check_metadata_deprecated(self):
         cmd = self._get_cmd()
         with check_warnings() as w:
             warnings.simplefilter("always")
+            warnings.filterwarnings("ignore", ".*OptionParser class will be replaced.*")
```
Comment (terryjreedy) [inline Lib/distutils/tests/test_register.py:294]: Add line for second Deprecation message added to issue report.
```suggestion
            warnings.filterwarnings("ignore", ".*OptionParser class will be replaced.*")
            warnings.filterwarnings("ignore", ".*Option class will be removed.*")
```

File: Lib/distutils/tests/test_sdist.py [hunk 2]
```diff
@@ -273,6 +273,7 @@ def test_check_metadata_deprecated(self):
         dist, cmd = self.get_cmd()
         with check_warnings() as w:
             warnings.simplefilter("always")
+            warnings.filterwarnings("ignore", ".*OptionParser class will be replaced.*")
```
Comment (terryjreedy) [inline Lib/distutils/tests/test_sdist.py]: Not needed
```suggestion
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]
pcbuild/build.bat -d  get compile warnings (separate issue)
```
f:\dev\311\Modules\_ssl.c(3624,28): warning C4244: 'function': conversion from
'uint64_t' to 'long', possible loss of data [f:\dev\311\PCbuild\_ssl.vcxproj]
f:\dev\311\Modules\_ssl.c(3638,12): warning C4244: '=': conversion from 'uint64
_t' to 'long', possible loss of data [f:\dev\311\PCbuild\_ssl.vcxproj]
```
Without the patch, test_distutils passes but with 7 link warnings like the following:
```
 Creating library build\temp.win-amd64-3.11-pydebug\Debug\xx_d.cp311-win_amd64.lib and object build\temp.win-amd64-3.11-pydebug\Debug\xx_d.cp311-win_amd64.exp
LINK : /LTCG specified but no code generation required; remove /LTCG from the link command line to improve linker performance
```
If CI and bots use this build, this explains why they pass.  I have seen these before; not sure if this test.

PCbuild/build.bat, without -d, supposed to be 'release' build, get same compile warnings, tests again pass.  Either this build is different from what is installed (perhaps the latter has pgo training?) or some other patch had effect.

In any case, after hand-editing the installed test_register to add your line and a similar line for a another deprecation message (see edit to issue report -- sorry I missed it before), tests pass on installed 3.11.8.  The edit to test_sdist is not needed.

I would apply the corrected patch rather than worry about why not needed for my local builds.

**Comment by serhiy-storchaka** [comment]
I have made the requested changes; please review again.

**Review by terryjreedy — APPROVED** [review]

**Comment by terryjreedy** [comment]
I verified that the two new lines in the patch are exactly equal to the two new lines in my installed copy.  Thank you for the fix.

**Pull request merged**